src/test/java/org/broadinstitute/consent/http/service/dao/DraftServiceDAOTest.java

```diff
@@ -0,0 +1,242 @@
+
+import com.google.cloud.storage.BlobId;
+import com.google.gson.Gson;
+import jakarta.ws.rs.BadRequestException;
+import jakarta.ws.rs.NotAuthorizedException;
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.StreamingOutput;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.IntStream;
+import org.broadinstitute.consent.http.cloudstore.GCSService;
+import org.broadinstitute.consent.http.db.DAOTestHelper;
+import org.broadinstitute.consent.http.enumeration.UserRoles;
+import org.broadinstitute.consent.http.models.DraftInterface;
+import org.broadinstitute.consent.http.models.DraftStudyDataset;
+import org.broadinstitute.consent.http.models.FileStorageObject;
+import org.broadinstitute.consent.http.models.User;
+import org.broadinstitute.consent.http.util.gson.GsonUtil;
+import org.glassfish.jersey.media.multipart.FormDataBodyPart;
+import org.jdbi.v3.core.statement.StatementException;
+import org.jdbi.v3.core.statement.UnableToExecuteStatementException;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class DraftServiceDAOTest extends DAOTestHelper {
+  @Mock
+  GCSService gcsService;
+
+  private DraftServiceDAO draftServiceDAO;
+
+  @BeforeEach
+  void setup() throws IOException {
+    DraftFileStorageServiceDAO draftFileStorageServiceDAO = new DraftFileStorageServiceDAO(jdbi, gcsService,
+        fileStorageObjectDAO);
+    this.draftServiceDAO = new DraftServiceDAO(jdbi, draftDAO,
+        draftFileStorageServiceDAO);
+  }
+
+  @Test
+  void testCreateDraft() throws SQLException, IOException {
+    when(gcsService.storeDocument(any(), anyString(), any())).thenReturn(
+        BlobId.of(UUID.randomUUID().toString(), UUID.randomUUID().toString()));
+    User user = createUser();
+    DraftInterface draft = createDraft(user, 3);
+    assertThat(draftDAO.findDraftsByUserId(user.getUserId()), hasSize(1));
+    Set<DraftInterface> storedDrafts = draftDAO.findDraftsByUserId(
+        user.getUserId());
+    assertThat(storedDrafts, hasSize(1));
+    DraftInterface storedDraft = storedDrafts.iterator().next();
+    assertThat(storedDraft.getStoredFiles(), hasSize(3));
+    assertEquals(storedDraft.getUUID(), draft.getUUID());
+  }
+
+  @Test
+  void testCreateDraftWithInvalidJson() {
+    User user = createUser();
+    DraftStudyDataset draft = new DraftStudyDataset("Hello world!",user);
+    assertThrows(BadRequestException.class, ()-> draftServiceDAO.insertDraft(draft));
```

Comment:
will be nice once we get spotless working in this repo
```suggestion
    assertThrows(BadRequestException.class, () -> draftServiceDAO.insertDraft(draft));
```